server/controllers/finance/reports/debtors/debtorAccountBalance.js

```diff
@@ -0,0 +1,91 @@
+const _ = require('lodash');
+const ReportManager = require('../../../../lib/ReportManager');
+const db = require('../../../../lib/db');
+
+// path to the template to render
+const TEMPLATE = './server/controllers/finance/reports/debtors/balance.handlebars';
+
+exports.debtorAccountBalance = debtorAccountBalance;
+
+
+/**
+ * @method debtorAccountBalance
+ *
+ * @description
+ * The HTTP interface which actually creates the report.
+ */
+function debtorAccountBalance(req, res, next) {
+  const options = _.extend(req.query, {
+    filename : 'REPORT.CLIENT_DEBTOR_ACCOUNT_BALANCE_REPORT',
+    orientation : 'portrait',
+    csvKey   : 'rows',
+    footerRight : '[page] / [toPage]',
+    footerFontSize : '7',
+  });
+
+  let report;
+  let data;
+
+  try {
+    report = new ReportManager(TEMPLATE, req.session, options);
+  } catch (e) {
+    return next(e);
+  }
+
+  // fire the SQL for the report
+
+  const fiscalYearId = req.params.fiscalYearId;
+  let label = '';
+  queryContext(fiscalYearId)
+  .then(
+    function(data) {
+
+      fiscalyerQuery(fiscalYearId).then(function (response){
+
+        if(response.length>0){
+          label = response[0].label;
+        }
+        var _data = {rows : data, fiscalLabel : label };
+        report.render(_data)
+        .then(result => {
+          res.set(result.headers).send(result.report);
+        })
+      })
+      .catch(next)
+      .done();
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+ * @method queryContext
+ *
+ * @param {Object} params Parameters passed in to customise the report - these
+ *                        are usually passed in through the query string
+ * @description
+ * The HTTP interface which actually creates the report.
+ */
+function queryContext(fiscalYearId) {
+
+  // aggregates the data above as totals into columns of 30, 60, 90, and >90
+  const sql = `
+    SELECT ac.id as account, ac.label as name, p.number as period,
+      SUM(pt.debit) as debit, SUM(pt.credit) as credit,
+      SUM(IF(p.number = 0,1,0))  as openingBalance,
+      SUM(IF(p.number <> 0, (pt.debit-pt.credit) ,0))  as closingBalance
+    FROM  period_total  pt
+    JOIN debtor_group d ON pt.account_id = d.account_id
+    JOIN account ac  ON ac.id = pt.account_id
+    JOIN fiscal_year f ON f.id = pt.fiscal_year_id
+    JOIN period p ON p.id = pt.period_id
+    WHERE pt.fiscal_year_id = ?
+    GROUP BY pt.account_id
+  `;
+  return db.exec(sql, fiscalYearId);
+}
+
+function fiscalyerQuery (id){
```

Comment:
This should probably be spelled `fiscalYearQuery`.

One of the most well known computer scientists, [Donald Knuth](https://en.wikipedia.org/wiki/Donald_Knuth), once remarked:
> Programs are meant to be read by humans and only incidentally for computers to execute.

This conveys the idea that your code should always be high quality for others to read, modify, and extend.  That is the reason we tend to be pretty strict on typos.